src/WebJobs.Script/Utility.cs

```diff
@@ -996,6 +997,56 @@ public static FunctionAppContentEditingState GetFunctionAppContentEditingState(I
             }
         }
 
+        public static void GetScaleInstancesToProcess(
+            IEnvironment environment,
+            IFunctionsHostingConfiguration hostingConfiguration,
+            IEnumerable<IScaleMonitor> scaleMonitors,
+            IEnumerable<ITargetScaler> targetScalers,
+            out List<IScaleMonitor> scaleMonitrsToProcess,
+            out List<ITargetScaler> targetScalersToProcess)
+        {
+            scaleMonitrsToProcess = new List<IScaleMonitor>();
+            targetScalersToProcess = new List<ITargetScaler>();
+
+            string value = environment.GetEnvironmentVariableOrDefault(EnvironmentSettingNames.TargetBaseScalingEnabled, "0");
+
+            // Check if TBS enabled on stamp level
+            if (value == "1")
+            {
+                string flags = hostingConfiguration.GetValue(ScriptConstants.ScaleControllerFeatureFlags, null).ToLower();
+                if (!string.IsNullOrEmpty(flags))
+                {
+                    foreach (var scaler in targetScalers)
+                    {
+                        if (flags.Contains(scaler.GetType().Name.ToLower()))
+                        {
+                            targetScalersToProcess.Add(scaler);
+                        }
+                    }
+                }
+
+                foreach (var monitor in scaleMonitors)
+                {
+                    // we made the assumption that if a listener implements both IScaleMonitor and ITargetScaler
+                    // IScaleMonitor.Descriptor.Id must start with "{functionId}-". For example: "{functionId}-QueueTrigger-{queueName}"
```

Comment:
This isn't a valid assumption. Note that the definition of the interface [here](https://github.com/Azure/azure-webjobs-sdk/blob/919d2d63419691166a6e36f1c1785f4c2c31aeb3/src/Microsoft.Azure.WebJobs.Host/Scale/ScaleMonitorDescriptor.cs#L17) indicates that this is just an opaque ID the format of which is completely determined by the extension.

Reply:
I was thinking it's not very good idea to use string keys. Type assembly name is much better.